<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Output files summary &mdash; TEFLoN2  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Config parameter file" href="../executing/configure_parameter_file.html" />
    <link rel="prev" title="Steps of TEFLoN2" href="steps_of_TEFLoN2.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> TEFLoN2
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/teflon2_tutorial.html">TEFLoN2 Tutorial</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Basic concepts</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="steps_of_TEFLoN2.html">Steps of TEFLoN2</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Output files summary</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#most-useful-output">Most useful output</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Executing</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../executing/configure_parameter_file.html">Config parameter file</a></li>
<li class="toctree-l1"><a class="reference internal" href="../executing/command_line_interface.html">Command line interface</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Project Info</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../project_info/citations.html">Citing and Citations</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">TEFLoN2</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Output files summary</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/basic_concepts/output_file_summary.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="output-files-summary">
<h1>Output files summary<a class="headerlink" href="#output-files-summary" title="Permalink to this heading"></a></h1>
<p>Here is the structure of the output files obtained after running the pipeline.</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>WORK_DIRECTORY/data_output_PREFIX/
├── 0-reference
│       ├── PREFIX.prep_MP
│       │       ├── PREFIX.annotatedTE.fa
│       │       ├── PREFIX.mappingRef.fa
│       │       ├── PREFIX.mappingRef.fa.amb
│       │       ├── PREFIX.mappingRef.fa.ann
│       │       ├── PREFIX.mappingRef.fa.bwt
│       │       ├── PREFIX.mappingRef.fa.pac
│       │       ├── PREFIX.mappingRef.fa.sa
│       │       └── PREFIX.pseudo.fa
│       ├── PREFIX.prep_TF
│       │   ├── PREFIX.genomeSize.txt
│       │   ├── PREFIX.hier
│       │   ├── PREFIX.pseudo2ref.txt
│       │   ├── PREFIX.ref2pseudo.txt
│       │   └── PREFIX.te.pseudo.bed
│       └── PREFIX.prep_RM             ----------
│           ├── GENOME.fasta                    |
│           ├── GENOME.fasta.align              |
│           ├── GENOME.fasta.cat.gz             |--- Specific to prepration custom
│           ├── GENOME.fasta.masked             |
│           ├── GENOME.fasta.out                |
│           ├── GENOME.fasta.tbl                |
│           └── PREFIX.bed             ----------
├── 1-mapping
│       ├── averageLength.all.txt
│       ├── SAMPLE_NAME.sorted.bam
│       ├── SAMPLE_NAME.sorted.bam.bai
│       ├── SAMPLE_NAME.sorted.cov.txt
│       ├── SAMPLE_NAME.sorted.stats.txt
│       ├── SAMPLE_NAME.sorted.subsmpl.bam
│       ├── SAMPLE_NAME.sorted.subsmpl.bam.bai
│       ├── SAMPLE_NAME.sorted.subsmpl.cov.txt
│       └── SAMPLE_NAME.sorted.subsmpl.stats.txt
├── 3-countPos
│       ├── SAMPLE_NAME.all_positions_sorted.collapsed.txt
│       ├── SAMPLE_NAME.all_positions_sorted.txt
│       ├── SAMPLE_NAME.all_positions.txt
│       ├── SAMPLE_NAME.counts.txt
│       ├── SAMPLE_NAME.pseudoSpace.genotypes.txt
│       ├── union_sorted.collapsed.txt
│       ├── union_sorted.txt
│       └── union.txt
├── 4-genotypes
│       ├── samples
│       │       ├── pseudoSpace
|       |       │       └── SAMPLE_NAME.pseudoSpace.genotypes.txt
|       |       ├── SAMPLE_NAME.genotypes.txt
|       |       ├── all_samples.genotypes.txt
|       |       └── all_samples.genotypes2.txt
|       └── populations                                  ----------
|           ├── NAME_POP.population.genotypes2.txt                |
|           ├── NAME_POP.population.genotypes.txt                 |--- If you use population file
│           ├── all_frequency.population.genotypes2.txt           |
│           └── all_frequency.population.genotypes.txt   ----------
└── sample_names.txt
</pre></div>
</div>
<section id="most-useful-output">
<h2>Most useful output<a class="headerlink" href="#most-useful-output" title="Permalink to this heading"></a></h2>
<p>The most useful output files are : WORK_DIRECTORY/data_output_PREFIX/4-genotypes//samples/SAMPLE_NAME.genotypes.txt</p>
<table class="docutils align-default" id="id1">
<caption><span class="caption-text">SAMPLE_NAME.genotypes.txt</span><a class="headerlink" href="#id1" title="Permalink to this table"></a></caption>
<colgroup>
<col style="width: 3.6%" />
<col style="width: 2.4%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
<col style="width: 7.2%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>chr</p></th>
<th class="head"><p>5’ breakpoint</p></th>
<th class="head"><p>3’ breakpoint</p></th>
<th class="head"><p>Level 1</p></th>
<th class="head"><p>Level 2</p></th>
<th class="head"><p>Stand</p></th>
<th class="head"><p>Reference TE ID</p></th>
<th class="head"><p>5’ soft-clipped reads</p></th>
<th class="head"><p>3’ soft-clipped reads</p></th>
<th class="head"><p>Presence reads</p></th>
<th class="head"><p>Abscence reads</p></th>
<th class="head"><p>Ambiguous reads</p></th>
<th class="head"><p>Genotype</p></th>
<th class="head"><p>Interpretation</p></th>
<th class="head"><p>Identifier TE</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>2R</p></td>
<td><p>38145</p></td>
<td><p>-</p></td>
<td><p>gate</p></td>
<td><p>ltr</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>1</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>1.0</p></td>
<td><p>present</p></td>
<td><p>te01</p></td>
</tr>
<tr class="row-odd"><td><p>2R</p></td>
<td><p>38145</p></td>
<td><p>-</p></td>
<td><p>rt1b</p></td>
<td><p>non-ltr</p></td>
<td><p>.</p></td>
<td><p>-</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>1</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>1.0</p></td>
<td><p>present</p></td>
<td><p>te02</p></td>
</tr>
<tr class="row-even"><td><p>2R</p></td>
<td><p>21076732</p></td>
<td><p>21077084</p></td>
<td><p>ine-1</p></td>
<td><p>helitron</p></td>
<td><p>-</p></td>
<td><p>FBti0061454</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>0</p></td>
<td><p>5</p></td>
<td><p>0</p></td>
<td><p>1.0</p></td>
<td><p>absent</p></td>
<td><p>te352</p></td>
</tr>
</tbody>
</table>
<ol class="arabic simple">
<li><p>chromosome</p></li>
<li><p>5’ breakpoint estimate (“-” if estimate not available)</p></li>
<li><p>3’ breakpoint estimate (“-” if estimate not available)</p></li>
<li><p>search level id (Usually TE family)</p></li>
<li><p>cluster level id (Usually TE order or class)</p></li>
<li><p>strand (“.” if strand could not be detected)</p></li>
<li><p>reference TE ID (“-” if novel insertion)</p></li>
<li><p>5’ breakpoint is supported by soft-clipped reads (if TRUE “+” else “-“)</p></li>
<li><p>3’ breakpoint is supported by soft-clipped reads (if TRUE “+” else “-“)</p></li>
<li><p>read count for “presence reads”</p></li>
<li><p>read count for “absence reads”</p></li>
<li><p>read count for “ambiguous reads”</p></li>
<li><p>genotype for every TE (allele frequency for pooled data, present/absent/no_data for haploid, present/absent/heterozygous/no_data for diploid)</p></li>
<li><p>frequency interpretation (for pooled data, present/absent/no_data for haploid, present/absent/heterozygous/no_data for diploid)</p></li>
<li><p>numbered identifier for each TE in the population</p></li>
</ol>
<p>If you use population file, output file are :  WORK_DIRECTORY/data_output_PREFIX/4-genotypes//populations/SAMPLE_POP.genotypes.txt</p>
<table class="docutils align-default" id="id2">
<caption><span class="caption-text">SAMPLE_POP.genotypes.txt</span><a class="headerlink" href="#id2" title="Permalink to this table"></a></caption>
<colgroup>
<col style="width: 3.4%" />
<col style="width: 2.2%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>chr</p></th>
<th class="head"><p>5’ breakpoint</p></th>
<th class="head"><p>3’ breakpoint</p></th>
<th class="head"><p>Level 1</p></th>
<th class="head"><p>Level 2</p></th>
<th class="head"><p>Stand</p></th>
<th class="head"><p>Reference TE ID</p></th>
<th class="head"><p>5’ soft-clipped reads</p></th>
<th class="head"><p>3’ soft-clipped reads</p></th>
<th class="head"><p>Presence reads</p></th>
<th class="head"><p>Abscence reads</p></th>
<th class="head"><p>Ambiguous reads</p></th>
<th class="head"><p>No data sample</p></th>
<th class="head"><p>Genotype</p></th>
<th class="head"><p>Interpretation</p></th>
<th class="head"><p>Identifier TE</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>2R</p></td>
<td><p>38145</p></td>
<td><p>-</p></td>
<td><p>gate</p></td>
<td><p>ltr</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>3</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>1.0</p></td>
<td><p>present</p></td>
<td><p>te01</p></td>
</tr>
<tr class="row-odd"><td><p>2R</p></td>
<td><p>38145</p></td>
<td><p>-</p></td>
<td><p>rt1b</p></td>
<td><p>non-ltr</p></td>
<td><p>.</p></td>
<td><p>-</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>4</p></td>
<td><p>4</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>1.0</p></td>
<td><p>polymorphic</p></td>
<td><p>te02</p></td>
</tr>
<tr class="row-even"><td><p>2R</p></td>
<td><p>21076732</p></td>
<td><p>21077084</p></td>
<td><p>ine-1</p></td>
<td><p>helitron</p></td>
<td><p>-</p></td>
<td><p>FBti0061454</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>0</p></td>
<td><p>7</p></td>
<td><p>0</p></td>
<td><p>1</p></td>
<td><p>1.0</p></td>
<td><p>absent</p></td>
<td><p>te352</p></td>
</tr>
</tbody>
</table>
<ol class="arabic simple">
<li><p>chromosome</p></li>
<li><p>5’ breakpoint estimate (“-” if estimate not available)</p></li>
<li><p>3’ breakpoint estimate (“-” if estimate not available)</p></li>
<li><p>search level id (Usually TE family)</p></li>
<li><p>cluster level id (Usually TE order or class)</p></li>
<li><p>strand (“.” if strand could not be detected)</p></li>
<li><p>reference TE ID (“-” if novel insertion)</p></li>
<li><p>5’ breakpoint is supported by soft-clipped reads (if TRUE “+” else “-“)</p></li>
<li><p>3’ breakpoint is supported by soft-clipped reads (if TRUE “+” else “-“)</p></li>
<li><p>read count for “presence reads”</p></li>
<li><p>read count for “absence reads”</p></li>
<li><p>read count for “ambiguous reads”</p></li>
<li><p>sample considered as no data</p></li>
<li><p>genotype for every TE</p></li>
<li><p>frequency interpretation (present/absent/polymorphic/no_data)</p></li>
<li><p>numbered identifier for each TE in the population</p></li>
</ol>
<p>If you use population file, output file are :  WORK_DIRECTORY/data_output_PREFIX/4-genotypes//populations/SAMPLE_POP.genotypes2.txt</p>
<table class="docutils align-default" id="id3">
<caption><span class="caption-text">SAMPLE_POP.genotypes2.txt</span><a class="headerlink" href="#id3" title="Permalink to this table"></a></caption>
<colgroup>
<col style="width: 3.4%" />
<col style="width: 2.2%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
<col style="width: 6.7%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>chr</p></th>
<th class="head"><p>5’ breakpoint</p></th>
<th class="head"><p>3’ breakpoint</p></th>
<th class="head"><p>Level 1</p></th>
<th class="head"><p>Level 2</p></th>
<th class="head"><p>Stand</p></th>
<th class="head"><p>Reference TE ID</p></th>
<th class="head"><p>5’ soft-clipped reads</p></th>
<th class="head"><p>3’ soft-clipped reads</p></th>
<th class="head"><p>Presence reads</p></th>
<th class="head"><p>Abscence reads</p></th>
<th class="head"><p>Ambiguous reads</p></th>
<th class="head"><p>No data sample</p></th>
<th class="head"><p>Genotype</p></th>
<th class="head"><p>Interpretation</p></th>
<th class="head"><p>Identifier TE</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>2R</p></td>
<td><p>38145</p></td>
<td><p>-</p></td>
<td><p>gate</p></td>
<td><p>ltr</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>4</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>1.0</p></td>
<td><p>present</p></td>
<td><p>te01</p></td>
</tr>
<tr class="row-odd"><td><p>2R</p></td>
<td><p>38145</p></td>
<td><p>-</p></td>
<td><p>rt1b</p></td>
<td><p>non-ltr</p></td>
<td><p>.</p></td>
<td><p>-</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>2</p></td>
<td><p>2</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>1.0</p></td>
<td><p>polymorphic</p></td>
<td><p>te02</p></td>
</tr>
<tr class="row-even"><td><p>2R</p></td>
<td><p>21076732</p></td>
<td><p>21077084</p></td>
<td><p>ine-1</p></td>
<td><p>helitron</p></td>
<td><p>-</p></td>
<td><p>FBti0061454</p></td>
<td><p>+</p></td>
<td><p>-</p></td>
<td><p>0</p></td>
<td><p>3</p></td>
<td><p>0</p></td>
<td><p>1</p></td>
<td><p>1.0</p></td>
<td><p>absent</p></td>
<td><p>te352</p></td>
</tr>
</tbody>
</table>
<ol class="arabic simple">
<li><p>chromosome</p></li>
<li><p>5’ breakpoint estimate (“-” if estimate not available)</p></li>
<li><p>3’ breakpoint estimate (“-” if estimate not available)</p></li>
<li><p>search level id (Usually TE family)</p></li>
<li><p>cluster level id (Usually TE order or class)</p></li>
<li><p>strand (“.” if strand could not be detected)</p></li>
<li><p>reference TE ID (“-” if novel insertion)</p></li>
<li><p>5’ breakpoint is supported by soft-clipped reads (if TRUE “+” else “-“)</p></li>
<li><p>3’ breakpoint is supported by soft-clipped reads (if TRUE “+” else “-“)</p></li>
<li><p>sample count considered as present</p></li>
<li><p>sample count considered as heterozygous</p></li>
<li><p>sample count considered as absent</p></li>
<li><p>sample considered as no data</p></li>
<li><p>genotype for every TE</p></li>
<li><p>frequency interpretation (present/absent/polymorphic/no_data)</p></li>
<li><p>numbered identifier for each TE in the population</p></li>
</ol>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="steps_of_TEFLoN2.html" class="btn btn-neutral float-left" title="Steps of TEFLoN2" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../executing/configure_parameter_file.html" class="btn btn-neutral float-right" title="Config parameter file" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Corentin Marco.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>